lib/redis/index.js

```diff
@@ -54,11 +55,36 @@ RedisClient.prototype.getLink = function(link) {
   })
 }
 
+RedisClient.prototype.fetchShortLinksFromTargetLinks = function(targetKeys) {
+  return Promise.all(targetKeys.map(targetKey =>
+    this.getShortLinksFromTargetLink(targetKey)
+  )).then(shortLinks =>
+    targetKeys.reduce((results, targetLinkKey, i) => {
+      results[Keys.getTargetLinkFromKey(targetLinkKey)] = shortLinks[i]
+      return results
+    }, {})
+  )
+}
+
+RedisClient.prototype.getShortLinksFromTargetLink = function(targetLink) {
+  return new Promise((resolve, reject) => {
+    this.impl.lrange(targetLink, 0, -1, (err, shortLinks) => {
+      return err ? reject(err) : resolve(shortLinks)
+    })
+  })
+}
+
 RedisClient.prototype.getLinks = function(searchString) {
```

Comment:
We should probably rename this to `searchShortLinks` to maintain parity with `searchTargetLinks`.